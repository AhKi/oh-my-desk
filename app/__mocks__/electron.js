export default jest.fn();

export const webContents = {
  getAllWebContents: jest.fn(() => []),
};

export const ipcMain = {
  on: jest.fn(),
};

export const BrowserWindow = jest.fn(info => ({
  close: jest.fn(),
  loadURL: jest.fn(),
  once: jest.fn(),
  on: jest.fn(),
  setAlwaysOnTop: jest.fn(),
  show: jest.fn(),
  showInactive: jest.fn(),
  getContentBounds: jest.fn(() => ({
    x: info.x,
    y: info.y,
    height: info.height,
    width: info.width,
  })),
  webContents: {
    on: jest.fn(),
    send: jest.fn(),
    openDevTools: jest.fn(),
  },
}));

export const ipcRenderer = {
  on: jest.fn(),
  send: jest.fn(),
};

export const remote = {
  getGlobal: jest.fn(),
  getCurrentWindow: jest.fn(),
  Menu: {
    buildFromTemplate: jest.fn(() => ({
      popup: jest.fn(),
    })),
  },
  app: {
    getVersion: jest.fn(),
  },
};

export const shell = {
  openExternal: jest.fn(),
};

export const app = {
  getName: jest.fn(() => 'oh-my-desk'),
  getPath: jest.fn(arg => arg),
  getVersion: jest.fn(),
  getAppPath: jest.fn(() => 'mock-path'),
  isReady: jest.fn(),
  on: jest.fn(),
};

export const Menu = {
  buildFromTemplate: jest.fn(() => ({
    popup: jest.fn(),
  })),
  setApplicationMenu: jest.fn(),
};

export const globalShortcut = {
  register: jest.fn(),
  unregister: jest.fn(),
};
