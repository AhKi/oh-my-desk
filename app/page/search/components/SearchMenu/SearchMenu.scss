.SearchMenu {
  background-color: #888888;
  float: left;
  height: 100%;
  width: 70px;
  text-align: center;
}

.SearchMenu__List {
  padding: 5px;
  margin: 5px;
}

.SearchMenu__Icon {
  width: 30px;
  margin: 5px;
}

.SearchMenu__Item {
  margin-top: 5px;
}

.SearchMenu__Btn {
  color: #e4e4e4;
  height: 100%;
  width: 100%;
}
