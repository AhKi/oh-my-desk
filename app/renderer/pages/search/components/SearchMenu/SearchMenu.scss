.SearchMenu {
  display: flex;
  flex-direction: column;
  background-color: #21242C;
  height: 100%;
}

.SearchMenu__Logo {
  padding: 12px 17px;
  align-self: center;
  background-color: #111111;

  svg {
    width: 35px;
  }
}

.SearchMenu__List {
  display: flex;
  flex-direction: column;
}

.SearchMenu__Item {
  height: 70px;
  background-color: #21242C;

  &:hover {
    background-color: #111111;
  }
}

.SearchMenu__Btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-size: 12px;
  color: #8c8c8c;

  &:hover {
    color: #ffffff;
  }
}

.SearchMenu__Btn--active {
  font-weight: 800;
  color: #ffffff;
}

.SearchMenu__Icon {
  align-self: center;
  vertical-align: middle;
  padding-bottom: 5px;

  svg {
    width: 25px;
  }
}


.SearchMenu__Btn {
  height: 100%;
  width: 100%;
}
