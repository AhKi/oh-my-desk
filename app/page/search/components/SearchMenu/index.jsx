import React from 'react';
// import PropTypes from 'prop-types';
import icon from 'assets/logo/logo-white.svg';
import addIcon from 'assets/icon/icon-widget-refresh.svg';
import allIcon from 'assets/icon/icon-menu-store.svg';
import favoritesIcon from 'assets/icon/icon-pin.svg';
import settingIcon from 'assets/icon/icon-menu-setting.svg';
import i18n from 'constants/i18n';
import './SearchMenu.scss';

const propTypes = {};
const defaultProps = {};

class SearchMenu extends React.Component {
  render() {
    const text = i18n().search;

    return (
      <div className="SearchMenu">
        <img
          className="SearchMenu__Logo"
          src={icon}
          alt="Oh-My-Desk-Logo"
        />
        <ul className="SearchMenu__List">
          <li className="SearchMenu__Item">
            <button
              className="SearchMenu__Btn"
              type="button"
            >
              <img
                className="SearchMenu__Icon"
                src={addIcon}
                alt=""
              />
              {text.newWidget}
            </button>
          </li>
          <li className="SearchMenu__Item">
            <button
              className="SearchMenu__Btn"
              type="button"
            >
              <img
                className="SearchMenu__Icon"
                src={allIcon}
                alt=""
              />
              {text.all}
            </button>
          </li>
          <li className="SearchMenu__Item">
            <button
              className="SearchMenu__Btn"
              type="button"
            >
              <img
                className="SearchMenu__Icon"
                src={favoritesIcon}
                alt=""
              />
              {text.favorites}
            </button>
          </li>
          <li className="SearchMenu__Item">
            <button
              className="SearchMenu__Btn"
              type="button"
            >
              <img
                className="SearchMenu__Icon"
                src={settingIcon}
                alt=""
              />
              {text.setting}
            </button>
          </li>
        </ul>
      </div>
    );
  }
}

SearchMenu.propTypes = propTypes;
SearchMenu.defaultProps = defaultProps;

export default SearchMenu;
