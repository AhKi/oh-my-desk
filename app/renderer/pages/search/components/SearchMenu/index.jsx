import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import icon from 'assets/oh-my-desk-icon.png';
import addIcon from 'assets/icon/icon-plus-white.svg';
import allIcon from 'assets/icon/icon-desktop-white.svg';
import favoritesIcon from 'assets/icon/icon-border-star-white.svg';
import outIcon from 'assets/icon/icon-out.svg';
import settingIcon from 'assets/icon/icon-menu-setting.svg';
import i18n from 'constants/i18n';
import './SearchMenu.scss';

const propTypes = {
  filter: PropTypes.string,
  onMakeWidgetRequest: PropTypes.func,
  onOpenPreference: PropTypes.func,
  onQuitApp: PropTypes.func,
  onSetFilter: PropTypes.func,
};
const defaultProps = {
  filter: 'ALL',
  onMakeWidgetRequest() {},
  onOpenPreference() {},
  onQuitApp() {},
  onSetFilter() {},
};

class SearchMenu extends React.Component {
  constructor(props) {
    super(props);
    this.handleSetAllFilter = this.handleSetAllFilter.bind(this);
    this.handleSetFavoritesFilter = this.handleSetFavoritesFilter.bind(this);
    this.handleMakeWidgetRequest = this.handleMakeWidgetRequest.bind(this);
  }

  handleMakeWidgetRequest() {
    const { onMakeWidgetRequest } = this.props;

    onMakeWidgetRequest();
  }

  handleSetAllFilter() {
    const { filter, onSetFilter } = this.props;

    if (filter !== 'ALL') {
      onSetFilter('ALL');
    }
  }

  handleSetFavoritesFilter() {
    const { filter, onSetFilter } = this.props;

    if (filter !== 'FAVORITES') {
      onSetFilter('FAVORITES');
    }
  }

  render() {
    const text = i18n().search;
    const { filter, onOpenPreference, onQuitApp } = this.props;
    const allMenuClassName = cx('SearchMenu__Btn', {
      'SearchMenu__Btn--active': filter === 'ALL',
    });
    const favoritesMenuClassName = cx('SearchMenu__Btn', {
      'SearchMenu__Btn--active': filter === 'FAVORITES',
    });

    return (
      <div className="SearchMenu">
        <img
          className="SearchMenu__Logo"
          src={icon}
          alt="Oh-My-Desk-Logo"
        />
        <ul className="SearchMenu__List">
          <li className="SearchMenu__Item">
            <button
              className="SearchMenu__Btn"
              type="button"
              onClick={this.handleMakeWidgetRequest}
            >
              <img
                className="SearchMenu__Icon"
                src={addIcon}
                alt=""
              />
              {text.newWidget}
            </button>
          </li>
          <li className="SearchMenu__Item">
            <button
              className={allMenuClassName}
              type="button"
              onClick={this.handleSetAllFilter}
            >
              <img
                className="SearchMenu__Icon"
                src={allIcon}
                alt=""
              />
              {text.all}
            </button>
          </li>
          <li className="SearchMenu__Item">
            <button
              className={favoritesMenuClassName}
              type="button"
              onClick={this.handleSetFavoritesFilter}
            >
              <img
                className="SearchMenu__Icon"
                src={favoritesIcon}
                alt=""
              />
              {text.favorites}
            </button>
          </li>
          <li className="SearchMenu__Item">
            <button
              className="SearchMenu__Btn"
              type="button"
              onClick={onOpenPreference}
            >
              <img
                className="SearchMenu__Icon"
                src={settingIcon}
                alt=""
              />
              {text.setting}
            </button>
          </li>
          <li className="SearchMenu__Item">
            <button
              className="SearchMenu__Btn"
              type="button"
              onClick={onQuitApp}
            >
              <img
                className="SearchMenu__Icon"
                src={outIcon}
                alt=""
              />
              {text.quit}
            </button>
          </li>
        </ul>
      </div>
    );
  }
}

SearchMenu.propTypes = propTypes;
SearchMenu.defaultProps = defaultProps;

export default SearchMenu;
