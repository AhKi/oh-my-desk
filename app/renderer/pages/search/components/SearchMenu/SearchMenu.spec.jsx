import { remote } from 'electron';
import React from 'react';
import { shallow } from 'enzyme';
import SearchMenu from '.';

describe('Test SearchMenu Component', () => {
  remote.getGlobal = jest.fn(() => () => JSON.stringify({ status: { lang: 'English' } }));
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match to snapshot', () => {
    const wrapper = shallow(<SearchMenu />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match to snapshot when props.filter === FAVORITES', () => {
    const wrapper = shallow(<SearchMenu filter="FAVORITES" />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('should test handleSetAllFilter', () => {
    it('when props.filter === FAVORITES', () => {
      const onSetFilter = jest.fn();
      const wrapper = shallow(
        <SearchMenu
          filter="FAVORITES"
          onSetFilter={onSetFilter}
        />,
      );

      wrapper.instance().handleSetAllFilter();

      expect(onSetFilter).toHaveBeenCalledTimes(1);
      expect(onSetFilter).toHaveBeenCalledWith('ALL');
    });

    it('when props.filter === ALL', () => {
      const onSetFilter = jest.fn();
      const wrapper = shallow(
        <SearchMenu
          filter="ALL"
          onSetFilter={onSetFilter}
        />,
      );

      wrapper.instance().handleSetAllFilter();

      expect(onSetFilter).toHaveBeenCalledTimes(0);
    });
  });

  describe('should test handleSetFavoritesFilter', () => {
    it('when props.filter === FAVORITES', () => {
      const onSetFilter = jest.fn();
      const wrapper = shallow(
        <SearchMenu
          filter="FAVORITES"
          onSetFilter={onSetFilter}
        />,
      );

      wrapper.instance().handleSetFavoritesFilter();

      expect(onSetFilter).toHaveBeenCalledTimes(0);
    });

    it('when props.filter === ALL', () => {
      const onSetFilter = jest.fn();
      const wrapper = shallow(
        <SearchMenu
          filter="ALL"
          onSetFilter={onSetFilter}
        />,
      );

      wrapper.instance().handleSetFavoritesFilter();

      expect(onSetFilter).toHaveBeenCalledTimes(1);
      expect(onSetFilter).toHaveBeenCalledWith('FAVORITES');
    });
  });
});
