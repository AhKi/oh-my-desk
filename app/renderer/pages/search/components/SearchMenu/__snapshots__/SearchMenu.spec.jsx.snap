// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchMenu Component should match to snapshot 1`] = `
<div
  className="SearchMenu"
>
  <SVGInline
    className="SearchMenu__Logo"
    classSuffix="-svg"
    cleanup={Array []}
    cleanupExceptions={Array []}
    component="span"
    svg=""
  />
  <ul
    className="SearchMenu__List"
  >
    <li
      className="SearchMenu__Item"
    >
      <button
        className="SearchMenu__Btn"
        data-test-id="menu-new-widget"
        onClick={[Function]}
        type="button"
      >
        <SVGInline
          className="SearchMenu__Icon"
          classSuffix="-svg"
          cleanup={Array []}
          cleanupExceptions={Array []}
          component="span"
          svg=""
        />
        New
      </button>
    </li>
    <li
      className="SearchMenu__Item"
    >
      <button
        className="SearchMenu__Btn SearchMenu__Btn--active"
        data-test-id="menu-all"
        onClick={[Function]}
        type="button"
      >
        <SVGInline
          className="SearchMenu__Icon"
          classSuffix="-svg"
          cleanup={Array []}
          cleanupExceptions={Array []}
          component="span"
          svg=""
        />
        All
      </button>
    </li>
    <li
      className="SearchMenu__Item"
    >
      <button
        className="SearchMenu__Btn"
        data-test-id="menu-favorites"
        onClick={[Function]}
        type="button"
      >
        <SVGInline
          className="SearchMenu__Icon"
          classSuffix="-svg"
          cleanup={Array []}
          cleanupExceptions={Array []}
          component="span"
          svg=""
        />
        Bookmark
      </button>
    </li>
    <li
      className="SearchMenu__Item"
    >
      <button
        className="SearchMenu__Btn"
        data-test-id="menu-setting"
        onClick={[Function]}
        type="button"
      >
        <SVGInline
          className="SearchMenu__Icon"
          classSuffix="-svg"
          cleanup={Array []}
          cleanupExceptions={Array []}
          component="span"
          svg=""
        />
        Options
      </button>
    </li>
    <li
      className="SearchMenu__Item"
    >
      <button
        className="SearchMenu__Btn"
        data-test-id="menu-quit"
        onClick={[Function]}
        type="button"
      >
        <SVGInline
          className="SearchMenu__Icon"
          classSuffix="-svg"
          cleanup={Array []}
          cleanupExceptions={Array []}
          component="span"
          svg=""
        />
        Quit
      </button>
    </li>
  </ul>
</div>
`;

exports[`Test SearchMenu Component should match to snapshot when props.filter === FAVORITES 1`] = `
<div
  className="SearchMenu"
>
  <SVGInline
    className="SearchMenu__Logo"
    classSuffix="-svg"
    cleanup={Array []}
    cleanupExceptions={Array []}
    component="span"
    svg=""
  />
  <ul
    className="SearchMenu__List"
  >
    <li
      className="SearchMenu__Item"
    >
      <button
        className="SearchMenu__Btn"
        data-test-id="menu-new-widget"
        onClick={[Function]}
        type="button"
      >
        <SVGInline
          className="SearchMenu__Icon"
          classSuffix="-svg"
          cleanup={Array []}
          cleanupExceptions={Array []}
          component="span"
          svg=""
        />
        New
      </button>
    </li>
    <li
      className="SearchMenu__Item"
    >
      <button
        className="SearchMenu__Btn"
        data-test-id="menu-all"
        onClick={[Function]}
        type="button"
      >
        <SVGInline
          className="SearchMenu__Icon"
          classSuffix="-svg"
          cleanup={Array []}
          cleanupExceptions={Array []}
          component="span"
          svg=""
        />
        All
      </button>
    </li>
    <li
      className="SearchMenu__Item"
    >
      <button
        className="SearchMenu__Btn SearchMenu__Btn--active"
        data-test-id="menu-favorites"
        onClick={[Function]}
        type="button"
      >
        <SVGInline
          className="SearchMenu__Icon"
          classSuffix="-svg"
          cleanup={Array []}
          cleanupExceptions={Array []}
          component="span"
          svg=""
        />
        Bookmark
      </button>
    </li>
    <li
      className="SearchMenu__Item"
    >
      <button
        className="SearchMenu__Btn"
        data-test-id="menu-setting"
        onClick={[Function]}
        type="button"
      >
        <SVGInline
          className="SearchMenu__Icon"
          classSuffix="-svg"
          cleanup={Array []}
          cleanupExceptions={Array []}
          component="span"
          svg=""
        />
        Options
      </button>
    </li>
    <li
      className="SearchMenu__Item"
    >
      <button
        className="SearchMenu__Btn"
        data-test-id="menu-quit"
        onClick={[Function]}
        type="button"
      >
        <SVGInline
          className="SearchMenu__Icon"
          classSuffix="-svg"
          cleanup={Array []}
          cleanupExceptions={Array []}
          component="span"
          svg=""
        />
        Quit
      </button>
    </li>
  </ul>
</div>
`;
